feat(popover): introduce disabled flag

medinarkhov opened the pull request:
Fixes #2188

PR created as a continuation of PR #2217 

`[disabled]` would be a useful feature for popover\toltip, allowing user to create conditional popover\tooltip.
`<div [ngbPopover]="It can be conditionally hidden!" [disabled]="hidePopover"></div>`

medinarkhov commented:
@pkozlowski-opensource please review. If the change looks good then I will add another commit to update:
-  tooltip to have `[disabled]` flag (remove conditional content then?)
- update API
- update demo

maxokorokov commented on src/popover/popover.ts:
```diff
@@ -115,6 +115,10 @@ export class NgbPopover implements OnInit, OnDestroy {
+  /**
+   * A flag indicating if a given popover is disabled and should not be displayed.
+   */
+  @Input() disabled = false;
```
1. Don't think it will be acceptable, too many collisions, ex:

```html
<button [disabled]="true" ngbPopover>...</button>
```

maybe `[showPopover]` or `[disablePopover]`?

```html
<div [ngbPopover]="text" [showPopover]="text !== ''" />
<div [ngbPopover]="text" [disablePopover]="text === ''" />
```

2. All other inputs initial values come from `NgbPopoverConfig`. I think we should put this one there as well. Not 100% sure though if there is a use case for massively initializing all popovers as disabled... So this is questionable

medinarkhov commented:
@maxokorokov thank you for prompt review. 

_Update:_
1. renamed `[disabled]` flag to `[disablePopover]`
2. updated `popover-config.ts`
3. won't touch Tooltip in this PR

API page looks good. Let me know if more changes required? Would you like demo example?

maxokorokov commented on src/popover/popover.ts:
```diff
@@ -115,6 +115,10 @@ export class NgbPopover implements OnInit, OnDestroy {
+  /**
+   * A flag indicating if a given popover is disabled and should not be displayed.
+   */
+  @Input() disablePopover = false;
```
You just forgot `this.disablePopover = config.disablePopover` in the constructor, and no need to initialize it here

medinarkhov replied:
good catch! fixed)

pkozlowski-opensource commented:
LGTM, thnx!
